extensions/metrics-cluster-feature/src/metrics-settings.tsx

```diff
@@ -0,0 +1,220 @@
+import React from "react";
+import { Component, Catalog, K8sApi } from "@k8slens/extensions";
+import { observer } from "mobx-react";
+import { observable } from "mobx";
+import { MetricsFeature, MetricsConfiguration } from "./metrics-feature";
+
+interface Props {
+  cluster: Catalog.KubernetesCluster;
+}
+
+@observer
+export class MetricsSettings extends React.Component<Props> {
+  @observable featureStates = {
+    prometheus: false,
+    kubeStateMetrics: false,
+    nodeExporter: false
+  };
+  @observable canUpgrade = false;
+  @observable upgrading = false;
+
+  config: MetricsConfiguration = {
+    prometheus: {
+      enabled: false
+    },
+    persistence: {
+      enabled: false,
+      storageClass: null,
+      size: "20G",
+    },
+    nodeExporter: {
+      enabled: false,
+    },
+    retention: {
+      time: "2d",
+      size: "5GB",
+    },
+    kubeStateMetrics: {
+      enabled: false,
+    },
+    alertManagers: null,
+    replicas: 1,
+    storageClass: null,
+  };
+  feature: MetricsFeature;
+
+  async componentDidMount() {
+    this.feature = new MetricsFeature(this.props.cluster);
+
+    await this.updateFeatureStates();
+  }
+
+  async updateFeatureStates() {
+    const status = await this.feature.getStatus();
+
+    this.canUpgrade = status.canUpgrade;
+    this.featureStates.prometheus = status.installed;
+
+    const deployment = K8sApi.forCluster(this.props.cluster, K8sApi.Deployment);
+
+    try {
+      await deployment.get({name: "kube-state-metrics", namespace: "lens-metrics"});
+      this.featureStates.kubeStateMetrics = true;
+    } catch(e) {
+      if (e?.error?.code === 404) {
+        this.featureStates.kubeStateMetrics = false;
+      } else {
+        this.featureStates.kubeStateMetrics = undefined;
+      }
+    }
+
+    const daemonSet = K8sApi.forCluster(this.props.cluster, K8sApi.DaemonSet);
+
+    try {
+      await daemonSet.get({name: "node-exporter", namespace: "lens-metrics"});
+      this.featureStates.nodeExporter = true;
+    } catch(e) {
+      if (e?.error?.code === 404) {
+        this.featureStates.nodeExporter = false;
+      } else {
+        this.featureStates.nodeExporter = undefined;
+      }
+    }
+  }
+
+  async save() {
+    this.config.prometheus.enabled = !!this.featureStates.prometheus;
+    this.config.kubeStateMetrics.enabled = !!this.featureStates.kubeStateMetrics;
+    this.config.nodeExporter.enabled = !!this.featureStates.nodeExporter;
+
+    if (!this.config.prometheus.enabled && !this.config.kubeStateMetrics.enabled && !this.config.nodeExporter.enabled) {
```

Comment:
User can delete/disable any of them. Problem with dedicated "install/delete" is the fact that "delete" would wipe all metrics history away. With toggles it's only happening if user chooses to disable "Prometheus" component.

Reason I chose to give freedom here: many times clusters might already have working metrics stack but it's configured so that Lens cannot query it properly. With this PR it's possible to just enable "Prometheus" which adds "Lens compatible" Prometheus service without the need to deploy duplicate node-exporter/kube-state-metrics (which would lead to duplicate metrics (2x cpu etc..))